src/core/lib/event_engine/thread_pool/thread_count.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include <grpc/support/port_platform.h>
+
+#include <atomic>
+#include <cstddef>
+#include <vector>
+
+#include "absl/base/thread_annotations.h"
+
+#include "src/core/lib/gprpp/sync.h"
+#include "src/core/lib/gprpp/time.h"
+
+namespace grpc_event_engine {
+namespace experimental {
+
+// Tracks counts across some fixed number of shards.
+// It is intended for fast increment/decrement operations, but a slower overall
+// count operation.
+class BusyThreadCount {
```

Comment:
Lets add a test for this. I'd suggest start some 10s of threads, increment/decrement a few 10's of thousands of times, and finally assert count == 0.

Reply:
Added. I also started a thread to concurrently call `count` while the inc/dec operations were happening.